timed/employment/views.py

```diff
@@ -345,6 +345,17 @@ class LocationViewSet(ReadOnlyModelViewSet):
     serializer_class = serializers.LocationSerializer
     ordering = ("name",)
 
+    def get_queryset(self):
+        """Don't show locations to customers."""
+        user = self.request.user
+
+        queryset = models.Location.objects.all()
+
+        if CustomerAssignee.objects.filter(user=user, is_customer=True).exists():
```

Comment:
Try to formulate the permissions in a "fail safe" way: In this variant, if there is no customer assignee, you can see everything. The goal should be that you see nothing UNLESS you have explicit permission, either via being a customer, or being an employee or external user with explicit permissions